tests/test_fx/test_complete_workflow.py

```diff
@@ -32,17 +59,45 @@ def forward(self, x):
         return x
 
 
-def run_workflow(world_size, dev):
+def GPT_config():
+    model_builder = lambda: GPTLMModel(hidden_size=1024, num_layers=24, num_attention_heads=16, checkpoint=True)
+    BATCH_SIZE = 1
+    SEQ_LENGTH = 16
+
+    def data_gen():
+        input_ids = torch.zeros((BATCH_SIZE, SEQ_LENGTH), dtype=torch.int64).to('meta')
+        attention_mask = torch.zeros((BATCH_SIZE, SEQ_LENGTH), dtype=torch.int64).to('meta')
+        kwargs = dict(input_ids=input_ids, attention_mask=attention_mask)
+        return kwargs
+
+    return model_builder, data_gen()
+
+
+def MLP_config():
+    return lambda: MLP(16), None
+
+
+MODEL_CONFIGS = [GPT_config()]
+
+
+## Randomly Generated Data
+def get_data(batch_size, seq_len, vocab_size):
+    input_ids = torch.randint(0, vocab_size, (batch_size, seq_len), device=torch.cuda.current_device())
+    attention_mask = torch.ones_like(input_ids)
+    return input_ids, attention_mask
+
+
+def run_workflow(world_size, dev, model_config):
     # initailization
     with LazyInitContext() as ctx:
-        model = MLP(16)
+        model = model_config[0]()
 
     for param in model.parameters():
         assert param.is_meta
 
     # tracing
     tracer = ColoTracer()
-    graph = tracer.trace(model)
+    graph = tracer.trace(model, meta_args=model_config[1])
     gm = torch.fx.GraphModule(model, graph, model.__class__.__name__)
 
     # annotate
```

Comment:
@YuliangLiu0306 I believe this function `transformer_mlp_pass` is not robust enough.

Reply:
Yes, it is a very simple pass which can only deals with simple MLP model. It was implemented only for the purpose of going through the workflow.

Reply:
After discussion with yuliang, he said `transformer_mlp_pass ` is a toy, so we will not use it in a real case.